core/aws-core/src/main/java/software/amazon/awssdk/awscore/endpointdiscovery/providers/EndpointDiscoveryProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.awscore.endpointdiscovery.providers;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+
+@FunctionalInterface
+@SdkInternalApi
+public interface EndpointDiscoveryProvider {
+
+    boolean resolveEndpointDiscovery();
```

Comment:
Can you add some docs here? I'm not sure what boolean is supposed to represent, I guess if endpoint discovery is enabled?